service/s3/internal/customizations/handle_200_error.go

```diff
@@ -0,0 +1,126 @@
+package customizations
+
+import (
+	"bytes"
+	"context"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/awslabs/smithy-go"
+	"github.com/awslabs/smithy-go/middleware"
+	smithyhttp "github.com/awslabs/smithy-go/transport/http"
+	smithyxml "github.com/awslabs/smithy-go/xml"
+)
+
+// HandleResponseErrorWith200Status check for S3 200 error response.
+// If an s3 200 error is found, status code for the response is modified temporarily to
+// 5xx response status code.
+func HandleResponseErrorWith200Status(stack *middleware.Stack) {
+	stack.Deserialize.Insert(&processResponseFor200ErrorMiddleware{}, "OperationDeserializer", middleware.After)
+	stack.Deserialize.Insert(&reset200StatusCodeMiddleware{}, "OperationDeserializer", middleware.Before)
```

Comment:
I don't think this is correct. The SDK should consider any 200 error as a 500 status code failure. swapping back to 200 can easily cause unexpected behavior.  Especially for the response error wrapper which maybe report a 200 error when the SDK should be reporting a 500 error back to the caller.

Reply:
To help clarify this a little. We should consider if it matters at all to inform the user that this was a 200 error vs just regular error.

Most likely the user doesn't need to know that a 200 was received but contained an error response. Most likely the user only needs to know the operation failed, the rest is internal details.

So with that said I suggest removing the 200 status reset middleware completely, and not replace it with anything.  Leave the rest of the middleware the way it is.